### PR TITLE
Added API props- contributors,links,redirects

### PR DESCRIPTION
**Pull request by abhigyank**
For issue #21 .
Added API properties -  contributors,links,redirects.
Changed _strip_JSON() parameters for different JSON response for different properties.

### CODE REVIEW

**Comment by abinashmeher999** [comment]
Hey @abhigyank, skimmed over it and it looks good! 💯 Please update your branch with the latest master. I will review it properly by today evening.

**Comment by abhigyank** [comment]
Updated branch with latest master.

File: sample/core.py [hunk 1]
```diff
@@ -49,11 +49,11 @@ def get_images(self, imlimit='max', imdir='ascending'):
 
         res = requests.get(self.base_url, params=self.payload, headers=self.headers).json()
 
-        img_list = _strip_JSON(res, prop, 'File')
+        img_list = _strip_JSON(res, prop, 'File','title')
```
Comment (abinashmeher999) [inline sample/core.py]: space after comma

File: sample/core.py [hunk 2]
```diff
@@ -69,16 +69,78 @@ def get_linkshere(self, lhprop="pageid|title|redirect", lhlimit="max"):
         lh_list = {}
         res = requests.get(self.base_url, params=self.payload,headers=self.headers).json()
 
-        lh_list = _strip_JSON(res, prop, '_nothing-to-strip_')
+        lh_list = _strip_JSON(res, prop, '_nothing-to-strip_','title')
```
Comment (abinashmeher999) [inline sample/core.py]: same as above

File: sample/core.py [hunk 3]
```diff
@@ -69,16 +69,78 @@ def get_linkshere(self, lhprop="pageid|title|redirect", lhlimit="max"):
...
 
+    # Method to fetch contributors
+
+    def get_contributors(self, pclimit="max"):
+        prop = 'contributors'
+        self.payload['prop'] = prop
+        self.payload['pclimit'] = pclimit
+
+        pc_list = {}
+        res = requests.get(self.base_url, params=self.payload,headers=self.headers).json()
```
Comment (abinashmeher999) [inline sample/core.py]: same as above

File: sample/core.py [hunk 4]
```diff
@@ -69,16 +69,78 @@ def get_linkshere(self, lhprop="pageid|title|redirect", lhlimit="max"):
...
+
+        pc_list = _strip_JSON(res, prop, '_nothing-to-strip_', 'name')
+        while 'continue' in res:
+            self.payload['pccontinue'] = res['continue']['pccontinue']
+            res = requests.get(self.base_url, params=self.payload,headers=self.headers).json()
```
Comment (abinashmeher999) [inline sample/core.py]: same here

File: sample/core.py [hunk 5]
```diff
@@ -69,16 +69,78 @@ def get_linkshere(self, lhprop="pageid|title|redirect", lhlimit="max"):
...
+            _append_results(pc_list, res, prop, '_nothing-to-strip_', 'name')
+
+        self.payload['pccontinue'] = None
+        self.payload['prop'] = None
+        return pc_list
+
+    # Method to fetch links
+
+    def get_links(self,pllimit="max", pldir="ascending"):
```
Comment (abinashmeher999) [inline sample/core.py]: and here

File: sample/core.py [hunk 6]
```diff
@@ -69,16 +69,78 @@ def get_linkshere(self, lhprop="pageid|title|redirect", lhlimit="max"):
...
+        prop = 'links'
+        self.payload['prop'] = prop
+        self.payload['pllimit'] = pllimit
+        self.payload['pldir'] = pldir
+        pl_list = {}
+        res = requests.get(self.base_url, params=self.payload,headers=self.headers).json()
```
Comment (abinashmeher999) [inline sample/core.py]: same. space needed after comma

File: sample/core.py [hunk 7]
```diff
@@ -69,16 +69,78 @@ def get_linkshere(self, lhprop="pageid|title|redirect", lhlimit="max"):
...
+
+        pl_list = _strip_JSON(res, prop, '_nothing-to-strip_', 'title')
+        while 'continue' in res:
+            self.payload['plcontinue'] = res['continue']['plcontinue']
+            res = requests.get(self.base_url, params=self.payload,headers=self.headers).json()
```
Comment (abinashmeher999) [inline sample/core.py]: here too

File: sample/core.py [hunk 8]
```diff
@@ -121,3 +183,6 @@ def _append_results(currlist, newlist, prop, strip_chars):
     pprint(wk.get_categories())
     pprint(wk.get_images())
     pprint(wk.get_linkshere())
+    pprint(wk.get_contributors())
+    pprint(wk.get_links())
+    pprint(wk.get_redirects())
```
Comment (abinashmeher999) [inline sample/core.py:194]: Also could you just print the name of the command before it actually prints. Easier that way to verify in the travis log.

File: sample/core.py [hunk 9]
```diff
@@ -69,16 +69,78 @@ def get_linkshere(self, lhprop="pageid|title|redirect", lhlimit="max"):
...
+        pc_list = {}
+        res = requests.get(self.base_url, params=self.payload,headers=self.headers).json()
+
+        pc_list = _strip_JSON(res, prop, '_nothing-to-strip_', 'name')
+        while 'continue' in res:
+            self.payload['pccontinue'] = res['continue']['pccontinue']
+            res = requests.get(self.base_url, params=self.payload,headers=self.headers).json()
+            _append_results(pc_list, res, prop, '_nothing-to-strip_', 'name')
+
+        self.payload['pccontinue'] = None
```
Comment (abinashmeher999) [inline sample/core.py]: I just noticed this. This has been followed for all the other methods. Ideally the key `pccontinue` and its value should be deleted from the dict after use in this method. Could you please do the same for the rest of the methods too in a separate commit?

**Review by abinashmeher999 — CHANGES_REQUESTED** [review]

File: sample/core.py (hunk 7 above)
Comment (abhigyank) [reply sample/core.py]: Is there any automated process to check this or do you just do manually?

Comment (abinashmeher999) [reply sample/core.py]: There are usually plugins for editors that do these checks according to PEP8. You can google if there are any for editor of your choice. For this PR I did manually.

**Review by abinashmeher999 — APPROVED** [review]

**Pull request merged**